### PR TITLE
Ability to handle a `PromiseObject` as value or selection parameter.

### LINKED ISSUE #20: Selection not set when using a RecordArray as `content` and a PromiseObject as `value`
I'm trying to build a select box to let the user choose a models.belongsTo property from a list of models retrieved through the store. While the selection process in general is working, the initial selection after reloading the page is not set.

`x-select content=recordArray selection=model.belongsToRecord optionValuePath="content" optionLabelPath="content.title"`

If I use `recordArray.content` and `model.belongsToRecord.content`, the initial selection is set properly, but the selection process stops working ( the model.belongsToRecord is not properly set anymore ).

Digging in with the Ember Inspector I can see that the value on the `x-select` component is a `PromiseObject` whereas the value of the `x-option` component is the resolved value of the Promise. 

I guess the comparison between those two with an `===` operator is not working.

### PR DESCRIPTION
**Pull request by mphasize**
Fixes #20

### CODE REVIEW

File: addon/components/x-option.js [hunk 1]
```diff
@@ -32,15 +32,22 @@ export default Ember.Component.extend({
...
+   *
    * @private
    * @property selected
    * @type Boolean
    */
   selected: Ember.computed('value', 'select.value', 'select.multiple', function() {
-    if (this.get('select.multiple') && isArray(this.get('select.value'))) {
-      return this.get('select.value').contains(this.get('value'));
+    var selectValue = this.get('select.value');
+    if(selectValue && selectValue.content) {
```
Comment (fivetanley) [inline addon/components/x-option.js]: What is content here?

Comment (mphasize) [inline addon/components/x-option.js]: http://emberjs.com/api/data/classes/DS.PromiseObject.html set's the `content` property once the Promise is resolved. Ideally it should be selectValue.then( fn(content) { ... }), but as far as I know ComputedProperties don't do async.

**Comment by cowboyd** [comment]
I'm very hesitant to put in a special case here for object proxying. Why not just use `content.content` as your value if that is indeed the real value?

**Comment by mphasize** [comment]
@cowboyd I understand the hesitancy and I am not sure, if this is the best solution. However, binding to `content.content` does not work. 

When I do that, the different `option`s still appear in the select menu, but initializing the selected option with a previously set value doesn't work. I tried all the possible combinations of handing data into the component, but in the end it seems to be an issue with observers not triggering properly/at the right time.

On the other hand, resolving the Promise manually triggers the change and fixes the initialization of the component.

**Comment by cowboyd** [comment]
It sounds really vexing, though it does make me thing that if initialization is not handled properly, then this it's either a framework bug or a bug in ember-data. In the event that we do allow for this, I'd rather have an adapter style interface where you can pass an adapter functions to get and set the selected value (as opposed to a special case for `PromiseObject`s.)

**Comment by mphasize** [comment]
@cowboyd How would such an adapter style interface look like? I don't know how the term applies here, so an example would be helpful. Thanks

**Comment by cowboyd** [comment]
Basically it means that you would delegate the responsibility of getting and setting the value to another object. It might look like:

```handlebars
{{x-select adapter='utils/x-select-promise-adapter' value=somePromiseProxy}}
<!-- ... -->
{{/x-select}}
```

Then in `utils/x-select-promise-adapter`, you'd do something approximating:

```javascript
export default Ember.Object.extend({
  dependencies: ['content.content']
  isSelected: function() {},
  read: function() {},
  write: function() {}
})
```
to provide guidance on how to detect if a value is selected, and how to read and write it. Although again, this feels like a bug and that the proxy is not actually reporting obvservation correctly.

**Comment by Robdel12** [comment]
Closing this due to inactivity. We can revisit if you'd like

**Comment by Asherlc** [comment]
I'd like to reopen this -- this was a rather confusing issue for me to sort out, and it comes up a lot for me when setting relationships, which I do often. Now that async relationships are the default in ember-data, I would venture to guess that this is a big use case, since simple string value selects can be easily done in regular HTMLBars.

The check is pretty easy, so it would be great if it could be incorporated natively.

**Pull request closed without merging**